core/camel-core/src/test/java/org/apache/camel/TestSupport.java

```diff
@@ -445,43 +422,6 @@ public static Channel unwrapChannel(Processor processor) {
         }
     }
 
-    /**
-     * Recursively delete a directory, useful to zapping test data
-     *
-     * @param file the directory to be deleted
-     */
-    public static void deleteDirectory(String file) {
-        deleteDirectory(new File(file));
-    }
-
-    /**
-     * Recursively delete a directory, useful to zapping test data
-     *
-     * @param file the directory to be deleted
-     */
-    public static void deleteDirectory(File file) {
-        if (file.isDirectory()) {
-            File[] files = file.listFiles();
-            if (files != null) {
-                for (File child : files) {
-                    deleteDirectory(child);
-                }
-            }
-        }
-
-        file.delete();
-    }
-
-    /**
-     * create the directory
-     *
-     * @param file the directory to be created
-     */
-    public static void createDirectory(String file) {
-        File dir = new File(file);
-        dir.mkdirs();
-    }
-
```

Comment:
We probably need to document the removal of these methods on the upgrade documentation. They may be used elsewhere / by users.

Reply:
Alternatively: wrapping them around the new `tempDir` stuff and marking them as deprecated would be acceptable as well (and, also, adding a note about their deprecation on the upgrade doc).

Reply:
That's a good point about not removing them and deprecating instead. I'll take this approach, and will also update the upgrade doc. Thanks!

Reply:
I added the deleted methods back and annotated as `@Deprecated`. About the doc, please let me know if I'm incorrect but this class `org.apache.camel.TestSupport` is from the `test` directory and shouldn't be visible to the end users of `camel-core`. Should I update 4.2→4.3 upgrade doc anyway?

Reply:
Ahh, good point. In this case we don't have to update the docs.